internal/internal.go

```diff
@@ -612,39 +620,121 @@ func getParsedPathAndQuery(req *ext_authz.CheckRequest) ([]interface{}, map[stri
 	return parsedPathInterface, parsedQueryInterface, nil
 }
 
-func getParsedBody(req *ext_authz.CheckRequest) (interface{}, bool, error) {
-	body := req.GetAttributes().GetRequest().GetHttp().GetBody()
+func getParsedBody(req *ext_authz.CheckRequest, parsedPath []interface{}, protoDescriptor string) (interface{}, bool, error) {
 	headers := req.GetAttributes().GetRequest().GetHttp().GetHeaders()
 
-	if body == "" {
-		return nil, false, nil
-	}
-
-	if val, ok := headers["content-length"]; ok {
-		cl, err := strconv.ParseInt(val, 10, 64)
-		if err != nil {
-			return nil, false, err
-		}
-
-		if cl != -1 && cl > int64(len(body)) {
-			return nil, true, nil
-		}
-	}
-
 	var data interface{}
 
 	if val, ok := headers["content-type"]; ok {
 		if strings.Contains(val, "application/json") {
+
+			body := req.GetAttributes().GetRequest().GetHttp().GetBody()
+
+			if body == "" {
+				return nil, false, nil
+			}
+
+			if val, ok := headers["content-length"]; ok {
+				cl, err := strconv.ParseInt(val, 10, 64)
+				if err != nil {
+					return nil, false, err
+				}
+				if cl != -1 && cl > int64(len(body)) {
+					return nil, true, nil
+				}
+			}
+
 			err := util.Unmarshal([]byte(body), &data)
 			if err != nil {
 				return nil, false, err
 			}
+		} else if strings.Contains(val, "application/grpc") {
+			rawbody := req.GetAttributes().GetRequest().GetHttp().GetRawBody()
+
+			if len(rawbody) <= 0 {
+				return nil, false, fmt.Errorf("invalid raw body")
+			}
+			if len(parsedPath) <= 1 {
+				return nil, false, fmt.Errorf("invalid parsed path")
+			}
+
+			if protoDescriptor == "" {
+				return nil, false, fmt.Errorf("ProtoDescriptor path not defined")
```

Comment:
❓Should this be an error? Imagine I'm using this and I don't care about authorizing gRPC at all. I wouldn't set this field. If a gRPC request was sent to opa-envoy-plugin then, should it not just ignore the body?

I'm thinking something along the lines of 
```go
		} else if strings.Contains(val, "application/grpc") {
			if protoDescriptor == "" {
				return nil, false, nil
			}
			// ...
}
```
Curious what you, and others, think about this case 😃

Reply:
it makes sense, its a way to turn off the the capability to decode the raw body